swiftide-integrations/src/pgvector/mod.rs

```diff
@@ -0,0 +1,163 @@
+    ///
+    /// Supports multiple field types; see [`FieldConfig`] for details.
+    #[builder(default)]
+    fields: Vec<FieldConfig>,
+}
+
+impl fmt::Debug for PgVector {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // Access the connection pool synchronously and determine the status.
+        let connection_status = self.connection_pool.connection_status();
+
+        f.debug_struct("PgVector")
+            .field("table_name", &self.table_name)
+            .field("vector_size", &self.vector_size)
+            .field("batch_size", &self.batch_size)
+            .field("connection_status", &connection_status)
+            .finish()
+    }
+}
+
+impl PgVector {
+    /// Creates a new instance of `PgVectorBuilder` with default settings.
+    ///
+    /// # Returns
+    ///
+    /// A new `PgVectorBuilder`.
+    pub fn builder() -> PgVectorBuilder {
+        PgVectorBuilder::default()
+    }
+
+    /// Retrieves a connection pool for `PostgreSQL`.
+    ///
+    /// This function returns the connection pool used for interacting with the `PostgreSQL` database.
+    /// It fetches the pool from the `PgDBConnectionPool` struct.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` that, on success, contains the `PgPool` representing the database connection pool.
+    /// On failure, an error is returned.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if it fails to retrieve the connection pool, which could occur
+    /// if the underlying connection to `PostgreSQL` has not been properly established.
+    pub fn get_pool(&self) -> Result<PgPool> {
+        self.connection_pool.get_pool()
+    }
+}
+
+impl PgVectorBuilder {
+    /// Tries to asynchronously connect to a `Postgres` server and initialize a connection pool.
+    ///
+    /// This function attempts to establish a connection to the specified `Postgres` server and
+    /// sets up a connection pool with an optional maximum number of connections.
+    ///
+    /// # Arguments
+    ///
+    /// * `url` - A string reference representing the URL of the `Postgres` server to connect to.
+    /// * `connection_max` - An optional value specifying the maximum number of connections in the pool.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` that contains an updated `PgVector` instance with the new connection pool on success.
+    /// On failure, an error is returned.
+    ///
+    /// # Errors
+    ///
+    /// This function returns an error if the connection to the database fails or if retries are exhausted.
+    /// Possible reasons include invalid database URLs, unreachable servers, or exceeded retry limits.
+    pub async fn try_connect_to_pool(
+        mut self,
+        url: impl AsRef<str>,
+        connection_max: Option<u32>,
+    ) -> Result<Self> {
+        let pool = self.connection_pool.clone().unwrap_or_default();
+
+        self.connection_pool = Some(pool.try_connect_to_url(url, connection_max).await?);
```

Comment:
See other comment. Changing that and moving it to the main struct will also remove the need to clone, unwrap and option.